CrossCompatibility/CrossCompatibility.psm1

```diff
@@ -0,0 +1,1744 @@
+[string]$script:AnyPlatformUnionPlatformName = [Microsoft.PowerShell.CrossCompatibility.Utility.PlatformNaming]::AnyPlatformUnionName
+[string]$script:AnyPlatformReferenceProfileFilePath = [System.IO.Path]::Combine($script:CompatibilityProfileDir, "$script:AnyPlatformUnionPlatformName.json")
+
+# Module path locations
+if ($PSVersionTable.PSVersion.Major -ge 6)
+{
+    [string]$script:PSHomeModulePath = [System.Management.Automation.ModuleIntrinsics].GetMethod('GetPSHomeModulePath', [System.Reflection.BindingFlags]'static,nonpublic').Invoke($null, @())
+    [string]$script:WinPSHomeModulePath = "$env:windir\System32\WindowsPowerShell\v1.0\Modules"
+}
+else
+{
+    [string]$script:PSHomeModulePath = "$PSHOME\Modules"
+    [string]$script:WinPSHomeModulePath = $script:PSHomeModulePath
+}
+
+<#
+.SYNOPSIS
+True if the given parameter name is a common cmdlet parameter, false otherwise.
+
+.PARAMETER ParameterName
+The cmdlet parameter name to test.
+#>
+function Test-IsCommonParameter
+{
+    param(
+        [Parameter(ValueFromPipeline=$true)]
+        [string]
+        $ParameterName
+    )
+
+    return $script:CommonParameters.Contains($ParameterName)
+}
+
+function Join-CompatibilityProfile
+{
+    [CmdletBinding(DefaultParameterSetName='File')]
+    param(
+        [Parameter(ParameterSetName='File', Position=0, ValueFromPipeline=$true)]
+        [string[]]
+        $InputFile,
+
+        [Parameter(ParameterSetName='Object', Position=0, ValueFromPipeline=$true)]
+        [Microsoft.PowerShell.CrossCompatibility.Data.CompatibilityProfileData[]]
+        $ProfileObject,
+
+        [Parameter()]
+        [string]
+        $ProfileId
+    )
+
+    if ($PSCmdlet.ParameterSetName -eq 'File')
+    {
+        $profiles = New-Object 'System.Collections.Generic.List[Microsoft.PowerShell.CrossCompatibility.Data.CompatibilityProfileData]]'
+
+        foreach ($path in $InputFile)
+        {
+            $resolvedPath = Resolve-Path $path
+
+            if (Test-Path $resolvedPath -PathType Container)
+            {
+                Get-ChildItem -Path $resolvedPath -Filter "*.json" `
+                    | ForEach-Object { ConvertFrom-CompatibilityJson -Path $_ } `
+                    | ForEach-Object { $profiles.Add($_) }
+
+                continue
+            }
+
+            $loadedProfile = ConvertFrom-CompatibilityJson -Path $resolvedPath
+            $profiles.Add($loadedProfile)
+        }
+
+        $ProfileObject = $profiles
+    }
+
+    return [Microsoft.PowerShell.CrossCompatibility.Utility.ProfileCombination]::UnionMany($ProfileId, $ProfileObject)
+}
+
+<#
+.SYNOPSIS
+Generate a new compatibility JSON file of the current PowerShell session
+at the specified location.
+
+.PARAMETER OutFile
+The file location where the JSON compatibility file should be generated.
+If this is null or empty, the result will be written to a file with a platform-appropriate name.
+
+.PARAMETER PassThru

```

Comment:
no help for the Validate, platformname, platformid, or readable parameters - I was confused when I saw the big validate function below